src/tools/illink/src/ILLink.Shared/DataFlow/ForwardDataFlowAnalysis.cs

```diff
@@ -155,14 +169,61 @@ public TState Get (TBlock block)
 		[Conditional ("DEBUG")]
 		public virtual void TraceBlockOutput (TValue normalState, TValue? exceptionState, TValue? exceptionFinallyState) { }
 
+		protected readonly TLattice lattice;
+
+		private readonly TValue entryValue;
+
+		protected ForwardDataFlowAnalysis (TLattice lattice, TValue entryValue)
+		{
+			this.lattice = lattice;
+			this.entryValue = entryValue;
+		}
+
+		void TransferOut (TTransfer transfer, TControlFlowGraph cfg, TBlock block, TState state,
+			Action<IControlFlowGraph<TBlock, TRegion>.ControlFlowBranch, TValue> updateState
+		)
+		{
+			TConditionValue? conditionValue = transfer.Transfer (block, state);
+
+			if (cfg.GetConditionalSuccessor (block) is IControlFlowGraph<TBlock, TRegion>.ControlFlowBranch conditionalBranch) {
+				// Duplicate the current state so that it's not shared with fall-through state.
+				TValue conditionalCurrentState = lattice.Meet (lattice.Top, state.Current);
+
+				if (conditionValue != null) {
```

Comment:
Vast majority of cases will have conditionValue==null (or should, feature checks should be very rare). Why do we need to duplicate the state for conditions which are not about feature checks?